src/tslint-cli.ts

```diff
@@ -243,8 +258,42 @@ const processFile = (file: string) => {
     }
 };
 
-const files = argv._;
+// if both files and tsconfig are present, use files
+let files = argv._;
+let program: ts.Program = undefined;
+
+if (argv.project != null) {
+    if (!fs.existsSync(argv.project)) {
+        console.error("Invalid option for project: " + argv.project);
+        process.exit(1);
+    }
+    program = Linter.createProgram(argv.project, path.dirname(argv.project));
+    if (files.length === 0) {
+        files = Linter.getFileNames(program);
+    }
+    if (argv["type-check"]) {
```

Comment:
I think it'd be nicer if we could detect whether we have any type based linter rules configured, and then throw an error here if type checking is not enabled. Otherwise we risk that users expect their type based linter rules run, but they are actually silently ignored.